code/modules/research/xenobiology/crossbreeding/_structures.dm

```diff
@@ -0,0 +1,667 @@
+	if(!ishuman(affected_mob))
+		return
+	var/mob/living/carbon/human/human_mob = affected_mob
+	human_mob.dna.remove_mutation(stored_mutation)
+
+/obj/structure/slime_crystal/pink
+	colour = "pink"
+
+/obj/structure/slime_crystal/pink/on_mob_enter(mob/living/affected_mob)
+	. = ..()
+	ADD_TRAIT(affected_mob,TRAIT_PACIFISM,MAGIC_TRAIT)
+
+/obj/structure/slime_crystal/pink/on_mob_leave(mob/living/affected_mob)
+	. = ..()
+	REMOVE_TRAIT(affected_mob,TRAIT_PACIFISM,MAGIC_TRAIT)
+
+/obj/structure/slime_crystal/gold
+	colour = "gold"
+
+/obj/structure/slime_crystal/gold/attack_hand(mob/user)
+	. = ..()
+	if(!ishuman(user))
+		return
+	var/mob/living/carbon/human/human_mob = user
+	var/mob/living/simple_animal/pet/chosen_pet = pick(subtypesof(/mob/living/simple_animal/pet))
+	chosen_pet = new chosen_pet(get_turf(human_mob))
+	human_mob.forceMove(chosen_pet)
+	human_mob.mind.transfer_to(chosen_pet)
+
+/obj/structure/slime_crystal/gold/on_mob_leave(mob/living/affected_mob)
+	. = ..()
+	if(!istype(affected_mob,/mob/living/simple_animal/pet))
+		return
+
+	var/mob/living/carbon/human/human_mob = locate() in affected_mob
+
+	if(!human_mob)
+		return
+
+	affected_mob.mind.transfer_to(human_mob)
+	human_mob.forceMove(get_turf(affected_mob))
+	qdel(affected_mob)
+
+/obj/structure/slime_crystal/oil
+	colour = "oil"
+
+/obj/structure/slime_crystal/oil/process()
+	for(var/T in RANGE_TURFS(3,src))
+		if(!isopenturf(T))
+			continue
+		var/turf/open/turf_in_range = T
+		turf_in_range.MakeSlippery(TURF_WET_LUBE,5 SECONDS)
+
+/obj/structure/slime_crystal/black
+	colour = "black"
+
+/obj/structure/slime_crystal/black/on_mob_effect(mob/living/affected_mob)
+	. = ..()
+	if(!ishuman(affected_mob) || isjellyperson(affected_mob))
+		return
+
+	if(affected_mobs[affected_mob] < 60) //Around 2 minutes
+		return
+
+	var/mob/living/carbon/human/human_transformed = affected_mob
+	human_transformed.set_species(pick(typesof(/datum/species/jelly)))
+
+/obj/structure/slime_crystal/lightpink
+	colour = "light pink"
+
+/obj/structure/slime_crystal/lightpink/attack_ghost(mob/user)
+	. = ..()
+	var/mob/living/simple_animal/hostile/lightgeist/L = new(get_turf(src))
+	L.ckey = user.ckey
+	affected_mobs[L] = 0
+
+/obj/structure/slime_crystal/lightpink/on_mob_leave(mob/living/affected_mob)
+	. = ..()
+	if(istype(affected_mob,/mob/living/simple_animal/hostile/lightgeist))
+		affected_mob.ghostize(TRUE)
+		qdel(affected_mob)
+
+/obj/structure/slime_crystal/adamantine
+	colour = "adamantine"
+
+/obj/structure/slime_crystal/adamantine/on_mob_enter(mob/living/affected_mob)
+	. = ..()
+	if(!ishuman(affected_mob))
+		return
+
+	var/mob/living/carbon/human/human = affected_mob
+	human.dna.species.brutemod *= 0.9
+	human.dna.species.burnmod *= 0.9
+
+/obj/structure/slime_crystal/adamantine/on_mob_leave(mob/living/affected_mob)
+	. = ..()
+	if(!ishuman(affected_mob))
+		return
+
+	var/mob/living/carbon/human/human = affected_mob
+	human.dna.species.brutemod /= 0.9
+	human.dna.species.burnmod /= 0.9
```

Comment:
Using - and + would be prefered, or use armour component since it handles removing better, otherwise things can get fucky if you remove 10% of their brutemod, something else like viro changes it by +4 and then you add 10% back on leaving the range resulting in more brutemod than you should have